js/dashboard-demo/tts.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Created by mgl on 03.06.19.
+ */
+
+/**
+ * Create a limit for the text to pronnounce.
+ * @type {number}
+ */
+const textToSpeechLimit = 30;
+/**
+ * Link of the end point.
+ * @type {Object}
+ */
+var socket;
+/**
+ * Text content and parametters.
+ * @type {Object}
+ */
+var msg;
+/**
+ * Synthetis voice API.
+ * @type {string}
+ */
+var synth;
+/**
+ * Voice parametters.
+ * @type {array}
+ */
+var voices;
+/**
+ * Text is actually playing or not.
+ * @type {bool}
+ */
+var speaking = false;
+/**
+ * Sound is actually playing or not.
+ * @type {bool}
+ */
+var playing = false;
+/**
+ * Sound content and parametters.
+ * @type {Object}
+ */
+var audio = new Audio();
+
+
+ /**
+  * Connects the user html page to the server,
+  * connects the client to WebSocket and
+  * subscribes client to notifications
+  * @constructor
+  */
+function connect() {
+  socket = new SockJS('http://mgladki.cern.ch:8080/fake-notifications');
+  stompClient = Stomp.over(socket);
+  stompClient.connect({}, function (frame) {
+    stompClient.subscribe('/topic/notification', function (content) {
+      var body = JSON.parse(content.body);
+      playOrDelay(body.filename, body.text);
+    });
+  });
+}
+
+/**
+ * Plays or not a notification if there
+ * is already a notification playing
+ * @param {string} filename The sound to play
+ * @param {string} text The text to pronnounce
+ */
+function playOrDelay(filename, text){
+  if(playing == false && speaking == false){
+    playSoundAndSpeak(filename, text);
+  }
+  else{
+    var wait;
+    wait = setTimeout(function() {playOrDelay(filename, text);}, 3000);
+  }
+}
+
+/**
+ * Plays at first the sound (with Playsound function)
+ * and then pronnounces the text (with textToSpeech function)
+ * @param {string} filename The sound to play
+ * @param {string} text The text to pronnounce
+ */
+function playSoundAndSpeak(filename, text){
+  if(filename === undefined){
+    textToSpeech(text);
+  }
+  else if (text === undefined){
+    playSound(filename);
+  }
+  else{
+    playSound(filename);
+    audio.onended = function(){
+      textToSpeech(text);
+    }
+  }
+}
+
+/**
+ * Play a sound
+ * @param {string} filename The sound to play
+ * @constructor
```

Comment:
This function is not a constructor - nor any function in this file. Please remove @constructor

Reply:
Ok for that one I wasn't sure